packages/maskbook/src/plugins/ITO/hooks/usePoolPayload.ts

```diff
@@ -1,6 +1,30 @@
-import { useAsyncRetry } from 'react-use'
+import { ValueRef } from '@dimensiondev/holoflows-kit'
 import { PluginITO_RPC } from '../messages'
+import { useValueRef } from '../../../utils/hooks/useValueRef'
+import type { JSON_PayloadInMask } from '../types'
+
+let payloadRef = new ValueRef<JSON_PayloadInMask | null>(null)
```

Comment:
It seems like all `PID`s are shared the same "cache". This is wrong, it will cause a race condition. If your data will never change or outdated, please reference my implementation: `getAssetAsBlobURL`. If it will change, please use https://github.com/vercel/swr (it supports suspense).

Reply:
I get it! thanks!

Reply:
Thanks, I get it.